internal/vds/vds.cpp

```diff
@@ -405,6 +408,80 @@ struct vdsbuffer fetch_slice_metadata(
     return buffer;
 }
 
+struct vdsbuffer fetch_fence(
+    const std::string& vds,
+    const std::string& credentials,
+    const std::string& coordinate_system,
+    const float* horizon,
+    size_t npoints
+) {
+    OpenVDS::Error error;
+    OpenVDS::VDSHandle handle = OpenVDS::Open(vds, credentials, error);
+
+    if (error.code != 0) {
+        throw std::runtime_error("Could not open VDS: " + error.string);
+    }
+
+    auto accessManager = OpenVDS::GetAccessManager(handle);
+    auto const *layout = accessManager.GetVolumeDataLayout();
+    const auto dimension_map =
+            layout->GetVDSIJKGridDefinitionFromMetadata().dimensionMap;
+
+    unique_ptr< float[][OpenVDS::Dimensionality_Max] > horizn(
+        new float[npoints][OpenVDS::Dimensionality_Max]{0}
+    );
+
+    auto coordinate_transformer = OpenVDS::IJKCoordinateTransformer(layout);
+    auto transform_coordinate = [&] (const float x, const float y) {
+        if (coordinate_system == "ij") {
+            return OpenVDS::Vector<double, 3> {x, y, 0};
+        }
+        else if (coordinate_system == "ilxl") {
+            return coordinate_transformer.AnnotationToIJKPosition({x, y, 0});
+        }
+        else if (coordinate_system == "cdp") {
+            return coordinate_transformer.WorldToIJKPosition({x, y, 0});
+        }
+        else {
+            const auto msg =
+                    "Coordinate system not recognized: " + coordinate_system;
+            throw std::runtime_error(msg);
+        }
+    };
+
+    for (size_t i = 0; i < npoints; i++) {
+        const float x = *(horizon++);
+        const float y = *(horizon++);
+
+        const auto coordinate = transform_coordinate(x, y);
+
+        horizn[i][dimension_map[0]] = coordinate[0];
+        horizn[i][dimension_map[1]] = coordinate[1];
+    }
+
+    // TODO: Verify that trace dimension is always 0
+    auto size = accessManager.GetVolumeTracesBufferSize(npoints, 0);
+
+    auto *data = new char[size];
```

Comment:
Yes, Erlend changed it after we had a discussion about it. It can actually cause a memory leak if an exception is thrown, so I will wrap it in a smart pointer like Erlend did.